### ISSUE TITLE
velero only processing one backup at a time

### ISSUE THREAD

**Issue by never1701** [vmware-tanzu/velero#2122]
**What steps did you take and what happened:**
velero configured to run 15 minutes backups on 94 namespaces.  Only one backup is processed at a time which fails after several hours with the following errors in the velero pod log: 

time="2019-12-11T18:20:15Z" level=info msg="1 errors encountered backup up item" backup=velero/fsdb-1316-20191210183938 group=v1 logSource="pkg/backup/resource_backupper.go:284" name=primary-86c585c894-sph6w namespace=fsdb-1316 resource=pods
time="2019-12-11T18:20:15Z" level=error msg="Error backing up item" backup=velero/fsdb-1316-20191210183938 error="timed out waiting for all PodVolumeBackups to complete" error.file="/go/src/github.com/vmware-tanzu/velero/pkg/restic/backupper.go:172" error.function="github.com/vmware-tanzu/velero/pkg/restic.(*backupper).BackupPodVolumes" group=v1 logSource="pkg/backup/resource_backupper.go:288" name=primary-86c585c894-sph6w namespace=fsdb-1316 resource=pods

restic log shows a slice out of bounds panic. 

**What did you expect to happen:**

For velero to be able to process more than one backup at a time.  

**The output of the following commands will help us better understand what's going on**:
(Pasting long output into a [GitHub gist](https://gist.github.com) or other pastebin is fine.)

https://gist.github.com/never1701/b6cc2adb842fc1fde0a2ee075c56cf69

* `kubectl logs deployment/velero -n velero`
* `velero backup describe <backupname>` or `kubectl get backup/<backupname> -n velero -o yaml`
* `velero backup logs <backupname>`
* `velero restore describe <restorename>` or `kubectl get restore/<restorename> -n velero -o yaml`
* `velero restore logs <restorename>`

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

**Environment:**

- Velero version (use `velero version`): 
v1.2
- Velero features (use `velero client config get features`): 
NOT SET
- Kubernetes version (use `kubectl version`):
Client Version: version.Info{Major:"1", Minor:"15", GitVersion:"v1.15.1", GitCommit:"4485c6f18cee9a5d3c3b4e523bd27972b1b53892", GitTreeState:"clean", BuildDate:"2019-07-18T09:18:22Z", GoVersion:"go1.12.5", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"14", GitVersion:"v1.14.6", GitCommit:"96fac5cd13a5dc064f7d9f4f23030a6aeface6cc", GitTreeState:"clean", BuildDate:"2019-08-19T11:05:16Z", GoVersion:"go1.12.9", Compiler:"gc", Platform:"linux/amd64"}

- Cloud provider or hardware configuration:
GCP

**Comment by never1701** [comment]
To add a little color, our environment worked fine for the first 40 or so namespaces, shortly around the time we got to the full 94 (we do still have others to add) we started seeing the number of backups being process drop until now it has dropped to one at a time.  Most of these pvs are actually quite small.  e.g. in this example fsdb-1316 is a pg data disk using 175MB out of 500GB disk.

**Comment by never1701** [comment]
Error from restic pod when backups eventually fail: 
time="2019-12-11T18:20:19Z" level=info msg="Setting --parent flag for this backup" backup=velero/fsdb-1969-20191211063138 controller=pod-volume-backup logSource="pkg/controller/pod_volume_backup_controller.go:249" name=fsdb-1969-20191211063138-hhp5m namespace=velero parentSnapshotID=96c63bfe
panic: runtime error: slice bounds out of range

goroutine 706 [running]:
github.com/vmware-tanzu/velero/pkg/restic.getLastLine(0xc000c46600, 0x0, 0x200, 0x0, 0xc000042001, 0xc000794f30)
        /go/src/github.com/vmware-tanzu/velero/pkg/restic/exec_commands.go:157 +0xcf
github.com/vmware-tanzu/velero/pkg/restic.RunBackup.func1(0xc00c379c80, 0x1d3e940, 0xc0007d1900, 0xc00c379c50, 0xc003cb78c0)
        /go/src/github.com/vmware-tanzu/velero/pkg/restic/exec_commands.go:99 +0x129
created by github.com/vmware-tanzu/velero/pkg/restic.RunBackup
        /go/src/github.com/vmware-tanzu/velero/pkg/restic/exec_commands.go:94 +0x153

**Comment by skriss** [comment]
@never1701 it's a known limitation that Velero only processes one backup at a time. #1531 covers running restic backups more concurrently, and #1653/#487 cover our thoughts on running Velero backups concurrently. Please add any further info you have there.

Will look into the panic around restic backups for this issue.

**Comment by never1701** [comment]
@skriss Thanks for the input, I will take a look at those issues and update if I have any further questions.  We have around 300 persistent volumes we are trying to backup and looking to determine if whole cluster or individual schedules are the best option.

**Comment by never1701** [comment]
Fixed this with a few changes to our set up, removed limits for restic pods, reinstalled velero, set the restic timeout high as well as starting over with an empty bucket and finally setting --snapshot-volumes=false.  Also we reverted to an hourly whole cluster backup and now things are completing without errors in a fairly short period of time.  3 hours for the first backup and around 15 - 30 minutes for the incrementals.
